DQM/Integration/python/clients/onlinebeammonitor_dqm_sourceclient-live_cfg.py

```diff
@@ -0,0 +1,179 @@
+                  VarParsing.VarParsing.varType.string,
+                  "Only lumisections with numbers matching lumiPattern are processed.")
+
+  options.register('eventsPerLumi',
+                  100,
+                  VarParsing.VarParsing.multiplicity.singleton,
+                  VarParsing.VarParsing.varType.int,
+                  "This number of last events in each lumisection will be processed.")
+
+  options.register('transDelay',
+                  0, #default value, int limit -3
+                  VarParsing.VarParsing.multiplicity.singleton,
+                  VarParsing.VarParsing.varType.int,
+                  "delay in seconds for the commit of the db transaction")
+
+  # This is used only by the online clients themselves. 
+  # We need to register it here because otherwise an error occurs saying that there is an unidentified option.
+  options.register('unitTest',
+                  True,
+                  VarParsing.VarParsing.multiplicity.singleton,
+                  VarParsing.VarParsing.varType.bool,
+                  "Required to avoid the error.")
+
+  options.register('noDB',
+                  True, # default value
+                  VarParsing.VarParsing.multiplicity.singleton,
+                  VarParsing.VarParsing.varType.bool,
+                  "Don't upload the BeamSpot conditions to the DB")
+
+  options.parseArguments()
+  
+  process.source = cms.Source("EmptySource")
+  process.source.numberEventsInRun=cms.untracked.uint32(100)
+  process.source.firstRun = cms.untracked.uint32(options.runNumber)
+  process.source.firstLuminosityBlock = cms.untracked.uint32(49)
+  process.source.numberEventsInLuminosityBlock = cms.untracked.uint32(2)
+  process.maxEvents = cms.untracked.PSet(
+              input = cms.untracked.int32(100)
+)
+
+else:
+  process.load("DQM.Integration.config.inputsource_cfi")
+  from DQM.Integration.config.inputsource_cfi import options
+  # for live online DQM in P5
+  # new stream label
+  process.source.streamLabel = cms.untracked.string('streamDQMOnlineBeamspot')
+
+#ESProducer
+process.load("CondCore.DBCommon.CondDBSetup_cfi")
+process.BeamSpotDBSource = cms.ESSource("PoolDBESSource",
+                      process.CondDBSetup,
+                      toGet = cms.VPSet(
+                            cms.PSet(
+                                record = cms.string('BeamSpotOnlineLegacyObjectsRcd'),
+                                tag = cms.string("BeamSpotOnlineTestLegacy"),
+                                refreshTime = cms.uint64(1)
+                            ),
+                            cms.PSet(
+                                record = cms.string('BeamSpotOnlineHLTObjectsRcd'),
+                                tag = cms.string("BeamSpotOnlineTestHLT"),
+                                refreshTime = cms.uint64(1)
+
+                            ),
+                      ),
+                      connect = cms.string('oracle://cms_orcon_prod/CMS_CONDITIONS')
```

Comment:
This won't work outside of P5 (e.g. in unit tests).
```
Monitoring file not found, disabling.
----- Begin Fatal Exception 23-Oct-2020 16:51:46 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESSource: class=PoolDBESSource label='BeamSpotDBSource'
Exception Message:
Service "cms_orcon_prod" can't be open with the current key. from cond::CredentialStore::setUpConnection 
----- End Fatal Exception -------------------------------------------------

---> test TestDQMOnlineClient-onlinebeammonitor_dqm_sourceclient had ERRORS
```

Reply:
but it is working on lxplus, at least it is working for me. Do you have the proper connection string to put?

Reply:
Maybe because you have been given a key file with reader access priviliges by @ggovi ? 
`frontier://FrontierProd/CMS CONDITIONS` will work for any user, but not sure that's what you want (at P5).